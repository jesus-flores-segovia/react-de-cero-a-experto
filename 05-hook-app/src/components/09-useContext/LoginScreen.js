import React, { useContext } from 'react';
import { useForm } from '../../hooks/useForm';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const [values, handleInputChange, reset] = useForm({
        email: "",
        password: "",
    });

    const {email, password} = values;

    const {user, setUser} = useContext(UserContext);

    const handleSubmit = (e) => {
        e.preventDefault();
        setUser(values);
        reset();
        console.log(values);
    };

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <form onSubmit={handleSubmit}>
                <div className="mb-3">
                    <label htmlFor="inputEmail" className="form-label">Email address</label>
                    <input name="email" type="email" className="form-control" id="inputEmail" aria-describedby="emailHelp" value={email} onChange={handleInputChange}/>
                </div>
                <div className="mb-3">
                    <label htmlFor="inputPassword" className="form-label">Password</label>
                    <input name="password" type="password" className="form-control" id="inputPassword" value={password} onChange={handleInputChange}/>
                </div>
                <button type="submit" className="btn btn-primary">Login</button>
            </form>
        </div>
    )
}
