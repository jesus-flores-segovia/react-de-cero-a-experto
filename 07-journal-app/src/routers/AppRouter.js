import { onAuthStateChanged } from 'firebase/auth'
import React, { useEffect, useState } from 'react'
import { useDispatch } from 'react-redux'
import { BrowserRouter, Route, Routes, Navigate } from 'react-router-dom'
import { login } from '../actions/auth'
import { startLoadingNotes } from '../actions/notes'

import { JournalScreen } from '../components/journal/JournalScreen'
import { auth } from '../firebase/firebaseConfig'
import { AuthRouter } from './AuthRouter'
import { PrivateRoute } from './PrivateRoute'
import { PublicRoute } from './PublicRoute'

export const AppRouter = () => {

    const dispatch = useDispatch();

    const [checking, setChecking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false)

    useEffect(() => {
        onAuthStateChanged(auth, (user) => {
            if(user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);

                dispatch(startLoadingNotes());
            } else{
                setIsLoggedIn(false);
            }

            setChecking(false);
        })
    }, [dispatch, setChecking, setIsLoggedIn])

    if(checking) {
        return (
            <h1>Wait...</h1>
        )
    }

    return (
        <BrowserRouter>
            <Routes>
                <Route path="/" 
                    element={
                        <PrivateRoute isLoggedIn={isLoggedIn}>
                            <JournalScreen/>
                        </PrivateRoute>
                        }/>
                <Route path="auth/*" 
                    element={
                        <PublicRoute isLoggedIn={isLoggedIn}>
                            <AuthRouter/>
                        </PublicRoute>
                        }/>
                <Route 
                    path="*"
                    element={<Navigate to="/auth/*" replace={true} />}
                />
            </Routes>
        </BrowserRouter> 
    )
}
