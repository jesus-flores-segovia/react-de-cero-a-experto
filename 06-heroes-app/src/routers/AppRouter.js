import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import { DcScreen } from "../components/dc/DcScreen";
import { LoginScreen } from "../components/login/LoginScreen";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { Navbar } from "../components/ui/NavBar";

export const AppRouter = () => {
    return (
        <Router>
          <div>
            <Navbar/>
                <Switch>
                    <Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/marvel" component={MarvelScreen}/>
                    <Route exact path="/dc" component={DcScreen}/>
                </Switch>
          </div>
        </Router>
      );
    }
