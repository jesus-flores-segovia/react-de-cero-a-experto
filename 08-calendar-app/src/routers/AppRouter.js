import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Navigate, Route, Routes } from "react-router-dom";
import { startCheckingLoginState } from "../actions/auth";
import { LoginScreen } from "../components/auth/LoginScreen";
import { CalendarScreen } from "../components/calendar/CalendarScreen";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {
  const dispatch = useDispatch();

  const { isLoggedIn, isCheckingLoginState } = useSelector(
    (state) => state.auth
  );

  useEffect(() => {
    dispatch(startCheckingLoginState());
  }, [dispatch]);

  if (isCheckingLoginState) {
    return <h1>Checking</h1>;
  }

  return (
    <Routes>
      <Route
        path="/"
        element={
          <PrivateRoute isLoggedIn={isLoggedIn} redirectPath={"login"}>
            <CalendarScreen />
          </PrivateRoute>
        }
      />
      <Route
        path="login"
        element={
          <PublicRoute isLoggedIn={isLoggedIn} redirectPath={"/"}>
            <LoginScreen />
          </PublicRoute>
        }
      />
      <Route path="login" element={<LoginScreen />} />
      <Route path="*" element={<Navigate to="/" />} />
    </Routes>
  );
};
