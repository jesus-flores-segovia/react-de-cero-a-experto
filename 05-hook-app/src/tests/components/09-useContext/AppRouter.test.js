import { mount } from "enzyme";
import { UserContext } from "../../../components/09-useContext/UserContext";
import { AppRouter } from "../../../components/09-useContext/AppRouter";

describe("Tests inside file 'AppRouter.test.js'", () => {

    const setUser = jest.fn();

    const user = {
        email: "floressegoviajesus@gmail.com",
        password: "12345"
    };

    const wrapper = mount(
        <UserContext.Provider value={{user, setUser}}>
            <AppRouter/>
        </UserContext.Provider>
    );

    test("'AppRouter' component must show correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

});