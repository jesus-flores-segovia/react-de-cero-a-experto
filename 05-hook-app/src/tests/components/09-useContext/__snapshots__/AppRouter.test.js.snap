// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file 'AppRouter.test.js' 'AppRouter' component must show correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "email": "floressegoviajesus@gmail.com",
   "password": "12345"
}
      </pre>
    </div>
  </div>
</div>
`;
